src/services/WolfDen.Domain/Entity/Employee.cs

```diff
@@ -1,54 +1,70 @@
+
 ﻿using EmployeeManagementDomain.Entity;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using static WolfDen.Domain.Enums.EmployeeEnum;
+
 
 namespace WolfDen.Domain.Entity
 {
     public class Employee
     {
         public int Id { get; private set; }
-        public int EmployeeId { get; private set; }
+        public int EmployeeCode { get; private set; }
         public string RFId { get; private set; }
-        public string FirstName { get; private set; }
+        public string? FirstName { get; private set; }
         public string? LastName { get; private set; }
-
-        public int? Age { get; private set; }
-        public string Email { get; private set; }
-        public string PhoneNumber { get; private set; }
+        public string? Email { get; private set; }
+        public string? PhoneNumber { get; private set; }
+        public DateOnly? DateofBirth { get; private set; }
 
         public DateOnly? JoiningDate { get; private set; }
-        public int DesignationId { get; private set; }
-        public Designation Designation { get; private set; }
-        public int DepartmentId {  get; private set; }
-        public Department Department { get; private set; }
+        public gender? Gender { get; private set; }
+        public int? DesignationId { get; private set; }
+        public Designation? Designation { get; private set; }
+        public int? DepartmentId {  get; private set; }
+        public Department? Department { get; private set; }
         public int? ManagerId { get; private set; }
-        public virtual Employee Manager {  get; private set; }
+        public virtual Employee? Manager {  get; private set; }
+        public bool? IsActive { get; private set; }
 
 
-        public Employee()
+        private Employee()
         {
 
         }
 
-        public Employee(int employeeId, string rfId, string firstName, string lastName, int age, string email, string phoneNumber, DateOnly joiningDate, int designationId,int departmentId,int managerId)
+        public Employee(int employeeCode, string rfId, string? firstName)
         {
-            EmployeeId = employeeId;
+            EmployeeCode = employeeCode;
             RFId = rfId;
             FirstName = firstName;
+
+            
+        }
+        public void EmployeeUpdateEmployee(string? firstName, string? lastName, DateOnly? dateofBirth, string? email, string? phoneNumber, gender? gender, DateOnly? joiningDate)
+        {
+            FirstName = firstName;
             LastName = lastName;
-            Age = age;
+            DateofBirth = dateofBirth;
             Email = email;
             PhoneNumber = phoneNumber;
+            Gender = gender;
             JoiningDate = joiningDate;
-            DesignationId = designationId;
+
+        }
+        public void AdminUpdateEmployee(int? designationId, int? departmentId, int? managerId, bool? isActive)
+        {
```

Comment:
Can't admin change the other employee details ?

Reply:
I think we don't need to maintain that restriction we can allow admin to edit all fields
since he have the ultimate control